new-installer/internal/steps/aws/rds_test.go

```diff
@@ -0,0 +1,263 @@
+					Value: json.RawMessage(`"postgres"`),
+				},
+				"password": {
+					Type:  json.RawMessage(`"string"`),
+					Value: json.RawMessage(`"fakepassword"`),
+				},
+			},
+		}, nil).Once()
+	}
+	if action == "upgrade" {
+		s.tfUtility.On("Run", mock.Anything, input).Return(steps.TerraformUtilityOutput{
+			TerraformState: "",
+			Output: map[string]tfexec.OutputMeta{
+				"host": {
+					Type:  json.RawMessage(`"string"`),
+					Value: json.RawMessage(`"rds-instance-12345678.us-west-2.rds.amazonaws.com"`),
+				},
+				"host_reader": {
+					Type:  json.RawMessage(`"string"`),
+					Value: json.RawMessage(`"rds-instance-reader-12345678.us-west-2.rds.amazonaws.com"`),
+				},
+				"port": {
+					Type:  json.RawMessage(`"string"`),
+					Value: json.RawMessage(`"5432"`),
+				},
+				"username": {
+					Type:  json.RawMessage(`"string"`),
+					Value: json.RawMessage(`"postgres"`),
+				},
+				"password": {
+					Type:  json.RawMessage(`"string"`),
+					Value: json.RawMessage(`"fakepassword"`),
+				},
+			},
+		}, nil).Once()
+
+		s.awsUtility.On("S3CopyToS3",
+			s.config.AWS.Region,
+			s.config.AWS.PreviousS3StateBucket,
+			fmt.Sprintf("%s/cluster/%s", s.config.AWS.Region, s.config.Global.OrchName),
+			s.config.AWS.Region,
+			s.config.Global.OrchName+"-"+s.runtimeState.DeploymentID,
+			"rds.tfstate",
+		).Return(nil).Once()
+
+		s.tfUtility.On("MoveStates", mock.Anything, steps.TerraformUtilityMoveStatesInput{
+			ModulePath: filepath.Join(s.step.RootPath, steps_aws.RDSModulePath),
+			States: map[string]string{
+				"module.aurora.aws_db_subnet_group.main":                "aws_db_subnet_group.main",
+				"module.aurora.aws_rds_cluster.main":                    "aws_rds_cluster.main",
+				"module.aurora.aws_rds_cluster_instance.main":           "aws_rds_cluster_instance.main",
+				"module.aurora.aws_rds_cluster_parameter_group.default": "aws_rds_cluster_parameter_group.default",
+				"module.aurora.aws_security_group.rds":                  "aws_security_group.rds",
+			},
+		}).Return(nil).Once()
+
+		s.tfUtility.On("RemoveStates", mock.Anything, steps.TerraformUtilityRemoveStatesInput{
+			ModulePath: filepath.Join(s.step.RootPath, steps_aws.RDSModulePath),
+			States: []string{
+				"module.s3",
+				"module.eks",
+				"module.efs",
+				"module.aurora_database",
+				"module.aurora_import",
+				"module.kms",
+				"module.orch_init",
+				"module.eks_auth",
+				"module.ec2log",
+				"module.aws_lb_controller",
+				"module.gitea",
+			},
+		}).Return(nil).Once()
+	}
+	if action == "uninstall" {
+		s.tfUtility.On("Run", mock.Anything, input).Return(steps.TerraformUtilityOutput{
+			TerraformState: "",
+			Output:         map[string]tfexec.OutputMeta{},
+		}, nil).Once()
+	}
+}
+
+func (s *RDSStepTest) expectAWSUtiliyCall(action string) {
+	if action == "install" {
+		s.awsUtility.On("GetAvailableZones", s.config.AWS.Region).
+			Return(availabilityZones, nil).
+			Times(2) // One for install, and one for uninstall
```

Comment:
Once? Since you already have the `else if` below for uninstall
```suggestion
			Once()
```